common/util/file_util.cc

```diff
@@ -69,6 +71,50 @@ static absl::Status CreateErrorStatusFromErrno(const char *fallback_msg) {
   }
 }
 
+// TODO (after bump to c++17) rewrite this function to use std::filesystem
+absl::Status UpwardFileSearch(absl::string_view start,
```

Comment:
Whlie playing with the thought of introducing `std::filesystem`, I noticed there is no unit-test for this function.

Background: I am considering that we should move forward with C++17; given that it is 2021, this sounds doable. (in fact, we have been compiling it for a while now as it was a temporary requirement for Kythe - nobody complained, but of course we need to double check that all releases on older platforms work.)

To be ready for more platform independent compilation (I am thinking of next frontier #307, now that MacOS is working), we can do that with `std::filesystem` and avoid Unix-specific calls. But of course prerequisite is to have unit tests to document the desired behavior...
(We need coverage ... #654 )

Reply:
What would be the best way to add unit tests for those? Setting up some fake virtual filesystem? Creating a tree in temp? We also need to make sure it's platform independent too

Reply:
Luckily we don't need a virtual filesystem, we can use the testing directory provided in the unit test framework with `testing::TempDir()`; that is a path provided by the sandboxed environment. We can create anything we want underneath.

Then use our own [`CreateDir()`](https://github.com/google/verible/blob/6ef4938d0a800b93ebf9fa7fdf692c4ed06c0471/common/util/file_util.h#L75) and  [`SetContents()`](https://github.com/google/verible/blob/6ef4938d0a800b93ebf9fa7fdf692c4ed06c0471/common/util/file_util.h#L69) functions to create a tree structure in the `testing::TempDir()`, and then test it with that.

Using that, we already have at least the test somewhat platform independent as we don't use platform specific functionality, only our own wrappers.